### DISCUSSION TITLE
Difficulty extracting historical build metrics from rsdoctor output

### DISCUSSION THREAD

**Discussion by grvdisco** [web-infra-dev/rsdoctor#627] [Q&A]
Hi team,

I’m trying to use rsdoctor to keep track of historical build metrics over time, including details like build time, bundle sizes, and duplicate packages. My goal is to store this information in a .csv file so I can plot trends and gain insights into how these parameters evolve across builds.

I’ve tried enabling the following config parameter:

`brief: {
  writeDataJson: true
}`

This resulted in a single manifest.json file that references a base64-compressed JSON file. After decompressing it, I still couldn’t derive meaningful metrics or data in a straightforward way that matches what the dashboard shows.

While exploring the rsdoctor codebase, I noticed that under the hood modern.js is used, along with an SDK that seems to be responsible for decompressing and translating the data into human-readable metrics. Unfortunately, I’m not familiar with modern.js, which makes it challenging to fully understand and extract these metrics.

Request:
Could you provide any guidance, documentation, or code references that explain how to properly decompress and interpret the rsdoctor data so I can map it to build metrics and store it externally (e.g., in a .csv file)? If there’s a known workflow or tool for this, I’d appreciate learning about it. And if there’s no existing solution, I’d be happy to try creating a PR—just point me in the right direction!

Thanks in advance for your help.

**Comment by yifancong** [comment]
Ok, I'll think about this and write a document for this situation this month.

**Reply by grvdisco** [reply]
Thanks! @easy1090 🖖

**Reply by grvdisco** [reply]
@easy1090 did you get a chance to write that document?

**Reply by yifancong** [reply]
> @easy1090 did you get a chance to write that document?

@grvdisco Ok, I have achieved this ability, but because there are a lot of tasks recently, I didn't sort out a document, which was delayed for a long time. You can use the `output.compressData`  this options in Version 1.0.0-beta.2. The document will be supported as soon as possible.

output.compressData can not compress data to support secondary acquisition and analysis.I don't know if it meets your expectations.

**Comment by yifancong** [comment]
You can use the output.compressData this options in Version 1.0.0-beta.2. The document will be supported as soon as possible.
output.compressData can not compress data to support secondary acquisition and analysis.I don't know if it meets your expectations.

https://rsdoctor.dev/config/options/options#compressdata
